xarray/core/accessor_str.py

```diff
@@ -2534,3 +2535,36 @@ def encode(self, encoding: str, errors: str = "strict") -> T_DataArray:
             encoder = codecs.getencoder(encoding)
             func = lambda x: encoder(x, errors)[0]
         return self._apply(func=func, dtype=np.bytes_)
+
+    def sig_str(self, n: int) -> DataArray:
```

Comment:
Again I think this is a too specific function.
Why not let the user enter the format string e.g.  `"{:2g}"`. Then the user is free to use whatever convention they like.

You can keep your example in the docs trying since I am sure it is the most common use case.

Also I think we need to come up with a better name for this function. Given that `da.str.format` already exists, I am not sure. This will be basically the same method but kind of reverse. Maybe `da.str.format_as`?

Reply:
If I get you properly, I should allow users to input the amount of significant figures and format they’ll like to use. I think that’s what `.str.format` does

Reply:
Yes basically `str.format` allows everything this methods already allows.
But apparently it is difficult to discover and/or use.
Maybe the "reverse" approach is simpler, so instead of `fmt.str.format(vals)` you could use `vals.str.format_as(fmt)` (name pending).

But definitely we should improve documentation on these methods.

Reply:
Sorry for the late reply because I’ve been busy working on different implementations and it’s just the same as using `.str.format`. What it uses is `vals.str.format(fmt)`. If you can give a small example on how you think our proposed method should work and the desired output, I think that would be very helpful.